gallery/src/DarkModeModal/DarkModeModal.tsx

```diff
@@ -0,0 +1,133 @@
+  const [mode, setMode] = useState<string | null>(null);
+  const [radioDisabled, setRadioDisabled] = useState(true);
+
+  const htmlElement = document.documentElement;
+  const htmlClasses = classnames('nx-html', 'nx-html--page-scrolling',
+      {'nx-html--enable-color-schemes': enableModeChange,
+        'nx-html--light-mode': mode === 'light' && enableModeChange,
+        'nx-html--dark-mode': mode === 'dark' && enableModeChange
+      });
+
+  useEffect(()=> {
+    htmlElement.className = htmlClasses;
+  }, [enableModeChange, mode]);
+
+  useEffect(()=> {
+    enableModeChange ? setRadioDisabled(false) : setRadioDisabled(true);
+  }, [enableModeChange]);
+
+  // const { classList } = document.documentElement;
+
+  // useEffect(()=>{
+  //   if (mode === 'light') {
+  //     classList.remove('nx-html--dark-mode');
+  //     classList.add('nx-html--light-mode');
+  //   }
+  //   else if (mode === 'dark') {
+  //     classList.remove('nx-html--light-mode');
+  //     classList.add('nx-html--dark-mode');
+  //   }
+  //   else {
+  //     classList.remove('nx-html--dark-mode', 'nx-html--light-mode');
+  //   }
+  // }, [enableModeChange, mode]);
+
+  // useEffect(()=> {
+  //   if (enableModeChange) {
+  //     classList.add('nx-html--enable-color-schemes');
+  //     setRadioDisabled(false);
+  //   }
+  //   else if (!enableModeChange) {
+  //     classList.remove('nx-html--enable-color-schemes', 'nx-html--dark-mode', 'nx-html--light-mode');
+  //     setRadioDisabled(true);
+  //   }
+  // }, [enableModeChange]);
+
+  return (
+    <>
+      <NxButton variant="tertiary" onClick={() => setShowModal(true)}>Light / Dark Mode</NxButton>
+      { showModal &&
+        <NxModal id="nx-modal-dark-mode-example"
+                 role="alertdialog"
+                 onCancel={modalCloseHandler}
+                 aria-label="NxModal to set light or dark mode">
+          <NxForm onSubmit={modalCloseHandler}
+                  submitBtnText= "Save and Close">
+            <header className="nx-modal-header">
+              <h2 className="nx-h2" id="modal-form-header">
+                <span>Setting Preferences for Light or Dark Mode</span>
+              </h2>
+            </header>
+            <div className="nx-modal-content">
+              <NxP>
+                In order to enable dark mode, the class <NxCode>nx-html--enable-color-schemes</NxCode> must first
+                be added to the <NxCode>html</NxCode> element, as demonstrated by the toggle below. Additional
+                classes <NxCode>nx-html--dark-mode</NxCode> or <NxCode>nx-html--light-mode</NxCode> may also be added
+                to force either light or dark mode, as demonstrated by the radio buttons. However, using these
+                simultaneously will result in dark mode and is not recommended. If neither class is added,
+                prefers-color-scheme in the user’s settings will dictate the mode.
```

Comment:
I'd like the explanation here to be understandable both to technical and non-technical people.  To do that, I think it should focus primarily on the conceptual mechanisms at play (e.g. that theming is opt-in and you always get light mode if you do not opt in, and also that when opted in your browser and/or OS theme choices determine the theme by default) but with brief mention of the technical implementations (classnames and media queries) as well.

Reply:
> I'd like the explanation here to be understandable both to technical and non-technical people. To do that, I think it should focus primarily on the conceptual mechanisms at play (e.g. that theming is opt-in and you always get light mode if you do not opt in, and also that when opted in your browser and/or OS theme choices determine the theme by default) but with brief mention of the technical implementations (classnames and media queries) as well.

As discussed in WW, the paragraph has been removed, and explanations implemented in tooltips here: [4e6b7de](https://github.com/sonatype/sonatype-react-shared-components/pull/782/commits/4e6b7deaa4d5e26ea2c2c25a0722bbe7ee68cbc8)